### PR TITLE
[Serverless] Run benchmarks on PR.

### PR DESCRIPTION
**Pull request by purple4reina**
### What does this PR do?

Adds a non-required, non-blocking github action run on pull request, but only when serverless code is updated.

Action runs benchmarks found in the serverless directory and posts the results as a comment to the PR. See the example comment below for what it will look like.

### Motivation

Customers care about overhead. Start and end invocation endpoints are directly corollated to the duration overhead customers see. Therefore, we should be keeping an eye on its performance over the long run.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by purple4reina** [comment]
> One comment, can we update the API to just edit it's own comment instead of pushing a new one and minimizing the old one? This can lead to 😠 🦄 in GitHub due to the amount of comments.
> 
> I'm saying this since I see multiple benchmark comments here, let me know if that's the case where we are already updating the one and only comment.

@duncanista good call. done.

File: .github/workflows/serverless-benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          cd go/src/github.com/DataDog/datadog-agent
+          benchstat -row /event previous current | tee analyze.txt
+          echo "analyze<<EOF" >> $GITHUB_OUTPUT
+          cat analyze.txt >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+      - name: Post comment
+        uses: marocchino/sticky-pull-request-comment@v2.5.0
+        with:
+          recreate: true
```
Comment (duncanista) [inline .github/workflows/serverless-benchmarks.yml:75]: amazing, just read the API for this

**Review by duncanista — APPROVED** [review]

File: .github/workflows/serverless-benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            </details>
+
+            <details open>
+            <summary>Benchmark stats</summary>
+
+            ```
+            ${{ steps.analyze.outputs.analyze }}
+            ```
+
+            </details>
```
Comment (ogaca-dd) [inline .github/workflows/serverless-benchmarks.yml:123]: Can you update the code owner for this file?

**Review by ogaca-dd — APPROVED** [review]

File: .github/workflows/serverless-benchmarks.yml (hunk 2 above)
Comment (purple4reina) [reply .github/workflows/serverless-benchmarks.yml:123]: Yup, already did!

**Review by blt — APPROVED** [review]
Neat.

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**